spring-cloud-dataflow-server-core/src/main/java/org/springframework/cloud/dataflow/server/support/DataFlowSwaggerApiDocsJsonDecodeFilter.java

```diff
@@ -0,0 +1,82 @@
+/*
+ * Copyright 2016-2018 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.springframework.cloud.dataflow.server.support;
+
+import java.io.IOException;
+import java.nio.charset.StandardCharsets;
+
+import javax.servlet.Filter;
+import javax.servlet.FilterChain;
+import javax.servlet.ServletException;
+import javax.servlet.ServletOutputStream;
+import javax.servlet.ServletRequest;
+import javax.servlet.ServletResponse;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletRequestWrapper;
+import javax.servlet.http.HttpServletResponse;
+
+import org.apache.commons.lang3.StringUtils;
+import org.apache.commons.text.StringEscapeUtils;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.core.annotation.Order;
+import org.springframework.stereotype.Component;
+import org.springframework.web.util.ContentCachingResponseWrapper;
+
+@Component
+@Order
+public class DataFlowSwaggerApiDocsJsonDecodeFilter implements Filter {
```

Comment:
See: https://github.com/spring-cloud/spring-cloud-dataflow/pull/4836#discussion_r817376744

Reply:
I had a look at the wrapper and they only store the original request / response but don't change anything to it. in the else block the original request and response is used for chaining. Because as for the if statements the default values of the properties are used I am sure we can consider it to be bullet-proof. wdyt?

Reply:
This has been solved by using the FilterRegistrationBean.